Dev/src/components/TopBanner.js

```diff
@@ -2,23 +2,61 @@
 //an icon, a title, and another icon that will all be equally seperated
 import FontLoadingComponent from './FontLoadingComponent';
 import React from 'React';
+import ImageSelectionModal from 'components/ImageSelectionModal'
+import strings from 'config/strings';
 import PropTypes from 'prop-types';
-import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import teacherImages from 'config/teacherImages'
+import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native';
 import { Icon } from 'react-native-elements';
 import colors from 'config/colors'
 import fontStyles from 'config/fontStyles';
 import { screenHeight, screenWidth } from 'config/dimensions';
+import { TextInput } from 'react-native-gesture-handler';
 
 class TopBanner extends FontLoadingComponent {
+    state={
+        modalVisible: false,
+    }
+    setModalVisible(visible) {
+        this.setState({ modalVisible: visible });
+    }
+    editProfilePic() {
+        this.setModalVisible(true);
+    }
+    onImageSelected(index) {
+        this.setState({ profileImageID: index, })
+        this.setModalVisible(false);
+    }
     render() {
         //Component properties
-        const { LeftIconName, LeftTextName, LeftOnPress, Title,
-            RightIconName, RightTextName, RightOnPress } = this.props;
+        const { LeftIconName, LeftTextName, LeftOnPress, Title, isEditingTitle, onTitleChanged,isEditingPicture,
+            RightIconName, RightTextName, RightOnPress, profilePic, UpdateProfileImage, profileImageID,} = this.props;
 
         return (
             <View style={styles.entireTopView}>
                 <View style={{ flex: 0.5 }} />
-                <View style={styles.topLeftView}  >
+                <View style={styles.topLeftView} >
+                    {(isEditingPicture ?
+                    <View>
+                    <ImageSelectionModal
+                        visible={this.state.modalVisible}
+                        images={teacherImages.images}
+                        cancelText={strings.Cancel}
+                        setModalVisible={this.setModalVisible.bind(this)}
+                        onImageSelected={this.onImageSelected.bind(this)}
+                    />
+                    <View style={styles.picContainer}>
+                        <Image
+                            style={styles.profilePic}
+                            onEditingPicture={UpdateProfileImage}
+                            source={teacherImages.images[profileImageID]} />
```

Comment:
You're setting the source of the image to the profileImageID that is from props, which means it cannot change. Make it this.state.profileImageID